cpp/examples/luma.gl/animometer.cc

```diff
@@ -109,53 +109,51 @@ auto createSampleData(int count) -> std::vector<ShaderData> {
   return shaderData;
 }
 
-}  // anonymous namespace
+auto makeWebGPUTable(wgpu::Device device) -> std::shared_ptr<garrow::Table> {
+  auto positionsDescriptor = garrow::AttributeDescriptor{"positions", arrow::fixed_size_list(arrow::float32(), 4)};
+  auto colorsDescriptor = garrow::AttributeDescriptor{"colors", arrow::fixed_size_list(arrow::float32(), 4)};
 
-struct Attribute {
-  const char* name;
-  void* data;
-  size_t bytes;
-};
+  auto positionsField = std::make_shared<garrow::Field>("positions", positionsDescriptor);
+  auto colorsField = std::make_shared<garrow::Field>("colors", colorsDescriptor);
 
-auto makeWebGPUTable(wgpu::Device device, const std::vector<Attribute>& attributes) {
-  std::vector<wgpu::Buffer> buffers;
-  for (auto const attribute : attributes) {
-    auto buffer = utils::createBufferFromData(device, attribute.data, attribute.bytes, wgpu::BufferUsage::Vertex);
-    buffers.push_back(buffer);
-  }
-  return buffers;
+  std::vector<std::shared_ptr<garrow::Field>> fields{positionsField, colorsField};
+  auto schema = std::make_shared<garrow::Schema>(fields);
+
+  std::vector<Vector4<float>> positionData{
+      {0.0f, 0.1f, 0.0f, 1.0f}, {-0.1f, -0.1f, 0.0f, 1.0f}, {0.1f, -0.1f, 0.0f, 1.0f}};
+  auto positionsArray = std::make_shared<garrow::Array>(device, positionsDescriptor, positionData);
+
+  std::vector<Vector4<float>> colorData{{1.0f, 0.0f, 0.0f, 1.0f}, {0.0f, 1.0f, 0.0f, 1.0f}, {0.0f, 0.0f, 1.0f, 1.0f}};
+  auto colorsArray = std::make_shared<garrow::Array>(device, colorsDescriptor, colorData);
```

Comment:
Do we need to pass the decriptor to the array AND the schema?

Reply:
Not really, descriptor in general is an intermediate solution I'm getting rid of now